tests/test_unit.py

```diff
@@ -1899,36 +1899,59 @@ def test_happy_path(self):
                 event = build_event(event_type, host)
                 headers = message_headers(event_type, host_id)
 
-                self.event_producer.write_event(event, host_id, headers)
+                self.event_producer.write_event(event, host_id, headers, wait=True)
 
-                send.assert_called_once_with(
-                    self.topic_name,
-                    key=host_id.encode("utf-8"),
-                    value=event.encode("utf-8"),
-                    headers=expected_encoded_headers(event_type, threadctx.request_id, host_id),
-                )
-                send.reset_mock()
+                produce.assert_called_once_with(self.topic_name, event.encode("utf-8"), callback=ANY)
+                flush.assert_called_once()
+
+                produce.reset_mock()
+                flush.reset_mock()
+
+    def test_producer_poll(self):
+        produce = self.event_producer._kafka_producer.produce
+        poll = self.event_producer._kafka_producer.poll
+        host_id = self.basic_host["id"]
+
+        for (event_type, host) in (
+            (EventType.created, self.basic_host),
+            (EventType.updated, self.basic_host),
+            (EventType.delete, deserialize_host(self.basic_host)),
+        ):
+            with self.subTest(event_type=event_type):
+                event = build_event(event_type, host)
+                headers = message_headers(event_type, host_id)
+
+                self.event_producer.write_event(event, host_id, headers, wait=False)
```

Comment:
`wait=False` is the default
```suggestion
                self.event_producer.write_event(event, host_id, headers)
```